src/CoreWCF.Primitives/src/CoreWCF/Security/DataProtectionSecurityStateEncoder.cs

```diff
@@ -3,64 +3,26 @@
 
 using System;
 using System.Security.Cryptography;
-using System.Text;
-using CoreWCF.Runtime;
+using Microsoft.AspNetCore.DataProtection;
 
 namespace CoreWCF.Security
 {
     public class DataProtectionSecurityStateEncoder : SecurityStateEncoder
     {
-        private readonly byte[] _entropy;
+        private readonly Lazy<IDataProtector> _dataProtector;
 
-        public DataProtectionSecurityStateEncoder() : this(true)
-        {
-            // empty
-        }
-
-        public DataProtectionSecurityStateEncoder(bool useCurrentUserProtectionScope) : this(useCurrentUserProtectionScope, null)
-        { }
-
-        public DataProtectionSecurityStateEncoder(bool useCurrentUserProtectionScope, byte[] entropy)
-        {
-            UseCurrentUserProtectionScope = useCurrentUserProtectionScope;
-            if (entropy == null)
-            {
-                _entropy = null;
-            }
-            else
-            {
-                _entropy = Fx.AllocateByteArray(entropy.Length);
-                Buffer.BlockCopy(entropy, 0, _entropy, 0, entropy.Length);
-            }
-        }
-
-        public bool UseCurrentUserProtectionScope { get; }
+        public DataProtectionSecurityStateEncoder() =>  _dataProtector = new Lazy<IDataProtector>(GetDataProtectorFromDI);
 
```

Comment:
This needs to stay, but can always return false.

Reply:
done